lib/requester/core.js

```diff
@@ -275,11 +277,25 @@ module.exports = {
             requestMethod = (typeof request.method === 'string') ? request.method.toLowerCase() : undefined,
             bodyIsEmpty = requestBody.isEmpty(),
             bodyIsDisabled = requestBody.disabled,
-            bodyContent = requestBody[requestBodyType];
+            bodyContent = requestBody[requestBodyType],
+            protocolProfileBehavior = requestBody.protocolProfileBehavior,
+            disableBodyPruning = protocolProfileBehavior === 'disable-body-pruning';
```

Comment:
Top level `var` for now, to be made a `const` once Node v4 support is dropped.
Also, this is always used in a negated form, so: `disableBodyPruning = protocolProfileBehaviour !== '...'`